libs/client/shared/src/hooks/useTeller.ts

```diff
@@ -0,0 +1,180 @@
+import { useEffect, useState } from 'react'
+import * as Sentry from '@sentry/react'
+import type { Logger } from '../providers/LogProvider'
+import toast from 'react-hot-toast'
+import { useTellerApi } from '../api'
+import type {
+    TellerConnectEnrollment,
+    TellerConnectFailure,
+    TellerConnectOptions,
+    TellerConnectInstance,
+} from 'teller-connect-react'
+import useScript from 'react-script-hook'
+type TellerEnvironment = 'sandbox' | 'development' | 'production' | undefined
+type TellerAccountSelection = 'disabled' | 'single' | 'multiple' | undefined
+const TC_JS = 'https://cdn.teller.io/connect/connect.js'
+
+// Create the base configuration for Teller Connect
+export const useTellerConfig = (logger: Logger) => {
+    return {
+        applicationId: process.env.NEXT_PUBLIC_TELLER_APP_ID ?? 'ADD_TELLER_APP_ID',
+        environment: (process.env.NEXT_PUBLIC_TELLER_ENV as TellerEnvironment) ?? 'sandbox',
+        selectAccount: 'disabled' as TellerAccountSelection,
+        onInit: () => {
+            logger.debug(`Teller Connect has initialized`)
+        },
+        onSuccess: {},
+        onExit: () => {
+            logger.debug(`Teller Connect exited`)
+        },
+        onFailure: (failure: TellerConnectFailure) => {
+            logger.error(`Teller Connect exited with error`, failure)
+            Sentry.captureEvent({
+                level: 'error',
+                message: 'TELLER_CONNECT_ERROR',
+                tags: {
+                    'teller.error.code': failure.code,
+                    'teller.error.message': failure.message,
+                },
+            })
+        },
+    } as TellerConnectOptions
+}
+
+// Custom implementation of useTellerHook to handle institution id being passed in
+export const useTellerConnect = (options: TellerConnectOptions, logger: Logger) => {
```

Comment:
This is lifted from the Teller github and modified so that I can pass in the provider id that is selected in the AccountTypeSelector component which then gets fed to the config before initializing the Teller Connect instance. It feels kinda janky, would love advice here.